extension/background.js

```diff
@@ -351,71 +354,78 @@ function showSurveyNotification(element, decision) {
-      case constants.EventType.SB_OTHER:
-      case constants.EventType.DOWNLOAD_MALICIOUS:
-      case constants.EventType.DOWNLOAD_DANGEROUS:
-      case constants.EventType.DOWNLOAD_DANGER_PROMPT:
-      case constants.EventType.EXTENSION_OTHER:
-        // Don't survey about these.
-        return;
-      case constants.EventType.UNKNOWN:
-        throw new Error('Unknown event type: ' + element['name']);
+  return new Promise(function(resolve, reject) {
+    chrome.storage.sync.get(cesp.READY_FOR_SURVEYS, function(items) {
+      if (!items[cesp.READY_FOR_SURVEYS]) return;
+      var userDecision = decision['name'];
+      if (userDecision !== constants.DecisionType.PROCEED &&
+          userDecision !== constants.DecisionType.DENY) {
+        reject();
         return;
-    }
-    getOperatingSystem().then(function(os) {
-      visitUrl = encodeURIComponent(visitUrl);
-      var openUrl = 'surveys/survey.html?js=' + surveyUrl + '&url=' + visitUrl
-          + '&os=' + os;
-      chrome.tabs.create(
-          {'url': chrome.extension.getURL(openUrl)},
-          function(tab) {
-            try {
-              chrome.tabs.remove(cesp.openTabId);
-            } catch (err) { }
-            cesp.openTabId = tab.id;
+      }
+
+      var surveyUrl, visitUrl;
+      var eventType = constants.FindEventType(element['name']);
+      switch (eventType) {
+        case constants.EventType.SSL_OVERRIDABLE:
+          surveyUrl = userDecision === constants.DecisionType.PROCEED ?
+              constants.SurveyLocation.SSL_OVERRIDABLE_PROCEED :
+              constants.SurveyLocation.SSL_OVERRIDABLE_NOPROCEED;
+          visitUrl = urlHandler.GetMinimalUrl(element['destination']);
+          break;
+        case constants.EventType.SSL_NONOVERRIDABLE:
+          surveyUrl = constants.SurveyLocation.SSL_NONOVERRIDABLE;
+          visitUrl = urlHandler.GetMinimalUrl(element['destination']);
+          break;
+        case constants.EventType.MALWARE:
+          surveyUrl = userDecision === constants.DecisionType.PROCEED ?
+              constants.SurveyLocation.MALWARE_PROCEED :
+              constants.SurveyLocation.MALWARE_NOPROCEED;
+          visitUrl = urlHandler.GetMinimalUrl(element['destination']);
+          break;
+        case constants.EventType.PHISHING:
+          surveyUrl = userDecision === constants.DecisionType.PROCEED ?
+              constants.SurveyLocation.PHISHING_PROCEED :
+              constants.SurveyLocation.PHISHING_NOPROCEED;
+          visitUrl = urlHandler.GetMinimalUrl(element['destination']);
+          break;
+        case constants.EventType.EXTENSION_INSTALL:
+        case constants.EventType.EXTENSION_INLINE_INSTALL:
+        case constants.EventType.EXTENSION_BUNDLE:
+          surveyUrl = userDecision === constants.DecisionType.PROCEED ?
+              constants.SurveyLocation.EXTENSION_PROCEED :
+              constants.SurveyLocation.EXTENSION_NOPROCEED;
+          break;
```

Comment:
thanks for checking. no, there's no visitUrl for extension installs, it's handled below w/o the visit url.